### PR TITLE
refactor: lock client api

### PR DESCRIPTION
**Pull request by LingKa28**
Base on #18 

Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
Refactor lock client api to improve easy of use of the kv clinet feature.

* what changes does this pull request make?
  * refactor lock clinet api & test sample

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
No.

### CODE REVIEW

File: client/lock.go [hunk 1]
```diff
@@ -251,9 +251,9 @@ func (c *lockClient) waitDelete(pfx string, myRev int64) {
 				}
 			}
 			if f {
-				cancel()
 				break
 			}
+			cancel()
```
Comment (bsbds) [inline client/lock.go:256]: Why do you move the cancel? And the `defer cancel()` in L242 seems useless.

File: client/lock.go [hunk 2]
```diff
@@ -251,9 +251,9 @@ func (c *lockClient) waitDelete(pfx string, myRev int64) {
 				}
 			}
 			if f {
```
Comment (iGxnon) [inline client/lock.go:253]: It is better a give `f` a full name.

Comment (iGxnon) [reply client/lock.go:253]: It seems you want break the outer for loop, try and a mark `outer` at L245 and use `break outer` ?

Comment (LingKa28) [reply client/lock.go:253]: You're right. I've made the changes you suggested.

File: client/lock.go [hunk 3]
```diff
@@ -242,18 +242,14 @@ func (c *lockClient) waitDelete(pfx string, myRev int64) {
...
 		defer cancel()
 		watcher, _ := c.watchClient.Watch(ctx, lastKey)
+	outer:
 		for r := range watcher {
-			f := false
 			for _, e := range r.Events {
 				if e.Type == xlineapi.Event_DELETE {
-					f = true
-					break
+					cancel()
```
Comment (bsbds) [inline client/lock.go:249]: There's already a defer on the cancel, so there's no need to call it explicitly

Comment (LingKa28) [reply client/lock.go]: Done.

**Review by bsbds — APPROVED** [review]

**Review by Phoenix500526 — APPROVED** [review]

**Pull request merged**